### PR TITLE
Update to UI forms system.

### PR DESCRIPTION
**Pull request by gopherstein**
Switched to a forms system that follows the
JSONschema standard better.

### CODE REVIEW

File: cmd/ui/assets/src/app/apps/apps-header/apps-header.component.ts [hunk 1]
```diff
@@ -76,7 +76,7 @@ export class AppsHeaderComponent implements OnDestroy, AfterViewInit {
               // TODO: Note - we need to add a sniffer here to look for a schema file
               // and any icon images in the chart. If they exist, we should use them instead of the generated one.
               this.appsModel.app.schema = GenerateSchema.json(this.appsModel.app.model);
-              (this.appsModel.app.schema as any).properties = this.setDescriptions((this.appsModel.app.schema as any).properties);
+              // (this.appsModel.app.schema as any).properties = this.setDescriptions((this.appsModel.app.schema as any).properties);
```
Comment (briansage) [inline cmd/ui/assets/src/app/apps/apps-header/apps-header.component.ts]: If this is unnecessary, would prefer to see this removed.
No need to comment out, as we have Git to keep in in history.

File: cmd/ui/assets/src/app/kubes/kubes.model.ts [hunk 2]
```diff
@@ -181,7 +185,9 @@ export class KubesModel {
         },
         'name': {
           'description': 'Name',
-          'type': 'string'
+          'type': 'string',
+          'pattern': '^[a-z]([-a-z0-9]*[a-z0-9])?$',
```
Comment (briansage) [inline cmd/ui/assets/src/app/kubes/kubes.model.ts]: Nice. Love this validation.

File: cmd/ui/assets/src/app/shared/edit-modal/edit-modal.component.html [hunk 3]
```diff
@@ -1,28 +1,35 @@
...
-            <button type="submit" class="btn btn-secondary" (click)="onSubmit()">{{ action }}</button>
+            <button type="submit" class="btn btn-secondary" (click)="onSubmit(value)">{{ action }}</button>
             <button type="button" class="btn btn-secondary" (click)="c('Close click')">Cancel</button>
           </div>
         </ng-template>
       </ngb-tab>
-      <ngb-tab title="Advanced" [disabled]="isDisabled">
+      <ngb-tab title="Advanced" [disabled]="isDisabled" select="setModel(value)">
         <ng-template ngbTabContent>
+<<<<<<< HEAD
```
Comment (briansage) [inline cmd/ui/assets/src/app/shared/edit-modal/edit-modal.component.html]: Ruh roh! Merge conflict!

File: cmd/ui/assets/src/app/kubes/kubes.model.ts (hunk 2 above)
Comment (gopherstein) [reply cmd/ui/assets/src/app/kubes/kubes.model.ts]: Weird... fixed your issues. Take a look.

Comment (gopherstein) [reply cmd/ui/assets/src/app/kubes/kubes.model.ts]: I think Atom messed up my commit somehow. I can't figure out how that merge conflict go pushed.

**Pull request merged**